### PR TITLE
Add GetEntryAndProof to log client.

### PR DESCRIPTION
**Pull request by Martin2112**

### CODE REVIEW

**Comment by pavelkalinnikov** [comment]
Heh, I just created the same method, was about to add a test :) Thanks for adding this.

File: client/logclient.go [hunk 1]
```diff
@@ -289,3 +289,21 @@ func (c *LogClient) GetAcceptedRoots(ctx context.Context) ([]ct.ASN1Cert, error)
 	}
 	return roots, nil
 }
+
+// GetEntryAndProof returns a log entry and audit path for the index of a leaf.
+func (c *LogClient) GetEntryAndProof(ctx context.Context, index uint64, treeSize uint64) (*ct.GetEntryAndProofResponse, error) {
```
Comment (pavelkalinnikov) [inline client/logclient.go]: Could shorten to `index, treeSize uint64`.

File: client/logclient_test.go [hunk 2]
```diff
@@ -712,3 +721,69 @@ func TestGetAcceptedRootsErrors(t *testing.T) {
...
+	}
+
+	for _, test := range tests {
+		resp, err := lc.GetEntryAndProof(context.Background(), test.index, test.treesize)
+		if err != nil {
+			t.Errorf("GetEntryAndProof(%v, %v)=nil,%v; want proof,nil", test.index, test.treesize, err)
+		} else if got := len(resp.AuditPath); got < 1 {
+			t.Errorf("len(GetEntryAndProof(%v, %v)): %v; want > 1", test.index, test.treesize, got)
+		}
+	}
```
Comment (pavelkalinnikov) [inline client/logclient_test.go]: Do you need to check validity of the entry and proof?

File: client/logclient_test.go [hunk 3]
```diff
@@ -712,3 +721,69 @@ func TestGetAcceptedRootsErrors(t *testing.T) {
...
+		{rsp: "", want: "EOF"},
+		{rsp: "not-json", want: "invalid"},
+		{rsp: `{"leaf_input": "bogus", "extra_data": "Z29vZAo=", "audit_path": ["Z29vZAo="]}`, want: "illegal base64"},
+		{rsp: `{"leaf_input": "Z29vZAo=", "extra_data": "bogus", "audit_path": ["Z29vZAo="]}`, want: "illegal base64"},
+		{rsp: `{"leaf_input": "Z29vZAo=", "extra_data": "Z29vZAo=", "audit_path": ["bogus"]}`, want: "illegal base64"},
+		{rsp: `{"leaf_input": "Z29vZAo=", "extra_data": "Z29vZAo=", "audit_path": ["bbbb",]}`, want: "invalid"},
+	}
+
+	for _, test := range tests {
+		ts := serveRspAt(t, "/ct/v1/get-entry-and-proof", test.rsp)
```
Comment (pavelkalinnikov) [inline client/logclient_test.go]: Maybe move this and `client.New` to above the loop?

File: client/logclient_test.go (hunk 2 above)
Comment (Martin2112) [reply client/logclient_test.go]: Dunno. I just copied the other tests in here.

File: client/logclient_test.go (hunk 3 above)
Comment (Martin2112) [reply client/logclient_test.go]: Cut and pasted.

File: client/logclient.go (hunk 1 above)
Comment (Martin2112) [reply client/logclient.go]: Done.

File: client/logclient_test.go (hunk 3 above)
Comment (pavelkalinnikov) [reply client/logclient_test.go]: Did?

Comment (Martin2112) [reply client/logclient_test.go]: No I left it the same as the other tests for the moment.

**Pull request merged**